nethttp/server_test.go

```diff
@@ -273,3 +273,59 @@ func BenchmarkStatusCodeTrackingOverhead(b *testing.B) {
 		}
 	})
 }
+
+func TestMiddlewareHandlerPanic(t *testing.T) {
+	recovery := func(f http.HandlerFunc) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+			defer func() {
+				if err := recover(); err != nil {
+					t.Logf("[Recovery] panic recovered\n")
+					w.WriteHeader(http.StatusInternalServerError)
+				}
+			}()
+			f(w, r)
+		}
+	}
+	tests := []struct {
+		handler         func(w http.ResponseWriter, r *http.Request)
+		spanFinishCount int
```

Comment:
Could consider removing spanFinishCount to be consistent with the other tests.

Reply:
OK, I will update the PR.